### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by haeminn**
I feel like there is a better way to write tests so I want some feedback before I write more.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
   "rulesDirectory": ["node_modules/codelyzer"],
-  "extends": ["codelyzer", "tslint-config-swimlane/base", "tslint-config-prettier"]
+  "extends": ["codelyzer", "tslint-config-swimlane/base", "tslint-config-prettier"],
+  "rules": {
+    "no-implicit-dependencies": [true, "dev"],
```
Comment (Hypercubed) [inline tslint.json]: Common is causing CI to fail...

File: projects/swimlane/ngx-ui/src/lib/components/input/input.component.spec.ts [hunk 2]
```diff
@@ -1,68 +1,200 @@
...
+      host.hostFixture.whenStable().then(() => {
+        host.hostFixture.detectChanges();
+        expect(host.element.querySelectorAll('.ngx-input-label span')[1].innerHTML).toEqual('***');
+      });
+    });
+
+    it('should autoselect', () => {
+      host = createHost(`<ngx-input [autoSelect]="true" [ngModel]="'hello'"></ngx-input>`);
+      expect(host.component.focused).toBeFalsy();
+      host.element.querySelector('input').dispatchEvent(new Event('focus'));
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/input/input.component.spec.ts]: Is this a work around?

**Comment by haeminn** [comment]
@Hypercubed Can you do a quick review of some changes I made? I fixed the failing calendar component test case.

**Comment by Hypercubed** [comment]
It all looks good to me... but I still don't undertsand the `host.element.querySelector('input').dispatchEvent(new Event('focus'));` line.

**Comment by haeminn** [comment]
That's how I was able to test `autoSelect`

**Pull request merged**